BUILDING.md

```diff
@@ -759,7 +759,156 @@ as `deps/icu` (You'll have: `deps/icu/source/...`)
+```
+This can be compiled using:
+```console
+$ make -j8
+$ make install_ssldirs
+```
+Next set the `PATH` environment variable to point to the `openssl` executable
+in the `apps` directory:
+```console
+$ export PATH=./apps:$PATH
+```
+Now that should be enough to run the target `install_fips` but there seems to
+be an error with alpha-15 in regards to this target (this should be fixed in
+alpha-16). This can be worked around by removing the last option of the openssl
+fipsinstall command so that is looks like this:
+<!--lint disable no-tabs-->
+```console
+install_fips: install_sw
+	@$(ECHO) "*** Installing FIPS module configuration"
+	@$(ECHO) "fipsinstall $(DESTDIR)$(MODULESDIR)/$(FIPSMODULENAME).cnf"
+	@openssl fipsinstall -module $(DESTDIR)$(MODULESDIR)/$(FIPSMODULENAME) \
+		-out $(DESTDIR)$(MODULESDIR)/$(FIPSMODULENAME).cnf
+```
+<!--lint disable no-tabs-->
+With those changes it should be possible to run the `install_fips` target:
+```console
+$ make install_fips
+```
+
+After the FIPS module and configuration file have been installed by the above
+instructions we also need to update `/path/to/install/dir/ssl/openssl.cnf` to
+use the generated FIPS configuration file (`fips.so.cnf`):
+```text
+.include /path/to/install/dir/lib/ossl-modules/fips.so.cnf
+
+# List of providers to load
+[provider_sect]
+default = default_sect
+# The fips section name should match the section name inside the
+# included fips.so.cnf.
+fips = fips_sect
+
+[default_sect]
+activate = 1
+```
+
+In the above case OpenSSL is not installed in the default location so two
+environment variables need to be set, `OPENSSL_CONF`, and `OPENSSL_MODULES`
+which should point to the OpenSSL configuration file and the directory where
+OpenSSL modules are located:
+```console
+$ export OPENSSL_CONF=/path/to/install/dir/ssl/openssl.cnf
+$ export OPENSSL_MODULES=/path/to/install/dir/lib/ossl-modules
+```
+
+Node.js can then be configured to enable FIPS:
+```console
+$ ./configure --shared-openssl --shared-openssl-libpath=/path/to/install/dir/lib --shared-openssl-includes=/path/to/install/dir/include --shared-openssl-libname=crypto,ssl --openssl-is-fips
+$ export LD_LIBRARY_PATH=/path/to/install/dir/lib
+$ make -j8
+```
+
+Verify the produced executable:
+```console
+$ ldd ./node
+	linux-vdso.so.1 (0x00007ffd7917b000)
+	libcrypto.so.81.3 => /path/to/install/dir/lib/libcrypto.so.81.3 (0x00007fd911321000)
+	libssl.so.81.3 => /path/to/install/dir/lib/libssl.so.81.3 (0x00007fd91125e000)
+	libdl.so.2 => /usr/lib64/libdl.so.2 (0x00007fd911232000)
+	libstdc++.so.6 => /usr/lib64/libstdc++.so.6 (0x00007fd911039000)
+	libm.so.6 => /usr/lib64/libm.so.6 (0x00007fd910ef3000)
+	libgcc_s.so.1 => /usr/lib64/libgcc_s.so.1 (0x00007fd910ed9000)
+	libpthread.so.0 => /usr/lib64/libpthread.so.0 (0x00007fd910eb5000)
+	libc.so.6 => /usr/lib64/libc.so.6 (0x00007fd910cec000)
+	/lib64/ld-linux-x86-64.so.2 (0x00007fd9117f2000)
+```
```

Comment:
Can also run `process.report.getReport()` in Node.js and look at the `sharedObjects` section to see loaded libraries.

Reply:
I'll add a note about using it. I think it might be useful to keep the `ldd` command just as a way to make sure the libraries can be found or node will note be able to run.

Reply:
+1 for keeping the `ldd` command. I'm pointing out an alternative... we could possibly also use the report in tests in the future as additional verification.